generate_workspace/src/test/java/com/google/devtools/build/workspace/maven/DefaultModelResolverTest.java

```diff
@@ -0,0 +1,71 @@
+package com.google.devtools.build.workspace.maven;
+
+import org.apache.maven.model.Repository;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import static com.google.common.truth.Truth.assertThat;
+
+/**
+ * Tests for {@link DefaultModelResolver}.
+ */
+@RunWith(JUnit4.class)
+public class DefaultModelResolverTest {
+  /** Test that addRepository doesn't replace existing repositories by default. */
+  @Test
+  public void testAddRepository_defaultReplace() {
+    Repository repoA1 = new Repository();
```

Comment:
First thank you for the tests. One nit

Since each test is effectively doing the same thing, can you create the following utility functions

```java
private Repository createRepository(String id, String url) {
     Repository repo = new Repository();
     repo.setId(id);
     repo.setUrl(url);
}

private void assertThatContainsRepo(
      DefaultModelResolver resolver, Repository repo) {...} 

private void assertThatUrlNotChanged(
       DefaultModelResolver resolver, Repository repo, String expectedUrl) {...}
```